simulator/src/bin/generate-drawing-examples.rs

```diff
@@ -21,64 +29,64 @@ macro_rules! op {
             $code;
 
             Ok::<(), core::convert::Infallible>(())
-        })().unwrap();
+        })()
+        .unwrap();
+
+        let cleansed_title = $title.replace(" ", "_").to_lowercase();
 
         let output_settings = OutputSettingsBuilder::new().scale(2).build();
 
-        let path = format!("{}/{}.png", OUTPUT_BASE, $title);
+        let file_path = format!("{}/{}.png", IMAGE_OUTPUT_BASE, cleansed_title);
         $display
             .to_image_buffer(&output_settings)
-            .save(&path)
+            .save(&file_path)
             .unwrap();
 
-        let screenshot = base64::encode(std::fs::read(&path).expect("Couldn't open screenshot"));
+        let doc_assets_path = format!("./assets/{}.png", cleansed_title);
 
         // Newlines in the code block aren't preserved by the stringify macro.
         // Use {} in the code block to insert newlines in the generated output.
         let doc_lines: Vec<_> = stringify!($code)
-                .trim_matches(|c| c == '{' || c == '}')
-                .trim()
-                .lines()
-                .map(|l| l.trim())
-                .map(|l| if l == "{ }" { "" } else { l }).collect();
+            .trim_matches(|c| c == '{' || c == '}')
+            .trim()
+            .lines()
+            .map(|l| l.trim())
+            .map(|l| if l == "{ }" { "" } else { l })
+            .collect();
 
-        // Note: empty lines must remain between HTML elements and inner Markdown for the Markdown
-        // to be parsed correctly.
         println!(
-            r#"//! ## {}
-            //!
-            //! {}
-            //!
-            //! <div style="display: flex">
-            //! <img style="width: 128px; height: 128px; margin-right: 8px;" alt="{} example screenshot" src="data:image/png;base64,{}" />
-            //! <div style="flex-grow: 1;">
+            r#"//!
+//! ## {}
+//!
+//! {}
+//!
+//! ![{} example screenshot]({})
```

Comment:
Can we keep the left alignment of the images? IMO that looked better.
According to this https://gist.github.com/DavidWells/7d2e0e1bc78f4ac59a123ddf8b74932d/ it should work by using an `img` tag with an `align` attribute.

Reply:
Sure, done. Much better than my old flexbox solution, which I don't think would've worked anyway as the inline styles are stripped by Github's markdown renderer. The only issue now is I can't get a margin between the image and code block:

![image](https://user-images.githubusercontent.com/800791/87559902-c6cca480-c6b2-11ea-93fc-eb890737f66f.png)

Doesn't look great, but it's not a major problem. I did try some inline `style=""` attributes but they're all stripped.

Reply:
Oh that's good to know! I've been using a Sublime Text plugin that I believe calls Github's API and shows the page standalone which I've always assumed is accurate to Github rendering. I guess there are some additional styles included on Github.com itself.